astroquery/esasky/core.py

```diff
@@ -1034,44 +1347,51 @@ def _extract_file_name_from_url(self, product_url):
         start_index = product_url.rindex("/") + 1
         return product_url[start_index:]
 
-    def _query_region(self, coordinates, radius, name, row_limit,
-                              get_query_payload, cache, json):
+    def _query(self, name, json, **kwargs):
         table_tap_name = self._find_mission_tap_table_name(json, name)
-        query = self._build_query(coordinates, radius, row_limit,
-                                          self._find_mission_parameters_in_json(table_tap_name,
-                                            json))
-        request_payload = self._create_request_payload(query)
-        if (get_query_payload):
-            return request_payload
-        return self._get_and_parse_from_tap(request_payload, cache)
+        if 'ids' in kwargs:
+            query = self._build_id_query(ids=kwargs.get('ids'),
+                                         row_limit=kwargs.get('row_limit'),
+                                         json=self._find_mission_parameters_in_json(table_tap_name, json))
+        else:
+            query = self._build_region_query(coordinates=kwargs.get('coordinates'), radius=kwargs.get('radius'),
+                                             row_limit=kwargs.get('row_limit'),
+                                             json=self._find_mission_parameters_in_json(table_tap_name, json))
+
+        if 'get_query_payload' in kwargs and kwargs.get('get_query_payload'):
+            return self._create_request_payload(query)
+
+        if not query:
+            return query
```

Comment:
Kindof. I've added an explanatory comment there plus a log in _build_id_query.

Reply:
In some other modules we raise an `InvalidQueryError` in cases like this, maybe that, or another Error type would make sense here, too and would be a quicker feedback for users that something is not right with their query. Anyway, this can be also a follow-up if you also think an error would make sense. 
(Either is fine by me at this point).